File: app-presentation/src/main/java/com/zxjdev/smile/presentation/common/base/activity/ActivityModule.java
package com.zxjdev.smile.presentation.common.base.activity;

import android.content.Context;

import com.zxjdev.smile.presentation.common.di.scope.ActivityScope;
import com.zxjdev.smile.presentation.common.util.ui.ErrorMessagePrinter;

import dagger.Module;
import dagger.Provides;

@Module
public class ActivityModule {

  @Provides
  @ActivityScope
  ErrorMessagePrinter provideErrorMessagePrinter(Context context) {
    return new ErrorMessagePrinter(context);
  }
}

File: app-presentation/src/main/java/com/zxjdev/smile/presentation/common/base/activity/ActivityModule.kt
package com.zxjdev.smile.presentation.common.base.activity

import android.content.Context

import com.zxjdev.smile.presentation.common.di.scope.ActivityScope
import com.zxjdev.smile.presentation.common.util.ui.ErrorMessagePrinter

import dagger.Module
import dagger.Provides

@Module
class ActivityModule {

    @Provides
    @ActivityScope
    internal fun provideErrorMessagePrinter(context: Context): ErrorMessagePrinter {
        return ErrorMessagePrinter(context)
    }
}
